### ISSUE TITLE
TIMESTAMP(p) datatype not parsed as single keyword in Trino dialect

### ISSUE THREAD

**Issue by rileymcdowell** [sqlfluff/sqlfluff#5589]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I wrote some Trino SQL with a `CAST(x AS TIMESTAMP(6))` in the query and linted it with SQLFluff. This generated an LT01 violation because SQLFluff wanted to see `TIMESTAMP (6)` (with a space). That's valid and equivalent, but an unusual formatting for a keyword that I don't think should be part of LT01 in Trino.

### Expected Behaviour

SQLFluff should recognize `TIMESTAMP(p)` as a single keyword token. Trino supports resolutions `(p)` from 0 (no fractional seconds) to 12 (picoseconds).

### Observed Behaviour

Here's an example SQL file:

```sql
SELECT CAST(DATE '2024-01-30' AS TIMESTAMP(6))
```

Here's my SQLFluff command:

```bash
sqlfluff lint --dialect trino ./query.sql
```

And here's the output:

```txt
== [query.sql] FAIL
L:   1 | P:  43 | LT01 | Expected single whitespace between 'TIMESTAMP' keyword
                       | and start bracket '('. [layout.spacing]
All Finished 📜 🎉!
```

### How to reproduce

Create a one-line SQL file called `query.sql` with the following contents:

```sql
SELECT CAST(DATE '2024-01-30' AS TIMESTAMP(6))
```

Then run the following command with the latest SQLFluff release:

```bash
sqlfluff lint --dialect trino ./query.sql
```

### Dialect

Trino

### Version

```
sqlfluff --version
sqlfluff, version 2.3.5
```
```bash
python --version
Python 3.9.16
```

### Configuration

No configuration file was used. This can be triggered with the default configuration.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
